src/deadline/job_attachments/api/attachment.py

```diff
@@ -28,56 +33,156 @@ def attachment_download(
     If path mapping rules file is given, map to corresponding destinations.
 
     Args:
-        manifests: File Path to the manifest file for upload.
-        s3_root_path: S3 root path including bucket name and root prefix.
-        boto_session: Boto3 session for interacting with customer s3.
-        path_mapping_rules: Optional file path to a list of path mapping.
+        manifests (List[str]): File Path to the manifest file for upload.
+        s3_root_uri (str): S3 root uri including bucket name and root prefix.
+        boto3_session (boto3.Session): Boto3 session for interacting with customer s3.
+        path_mapping_rules (Optional[str], optional): Optional file path to a JSON file contains list of path mapping. Defaults to None.
+        logger (ClickLogger, optional): Logger to provide visibility. Defaults to ClickLogger(False).
     """
 
-    # path validation
-    if not all([os.path.isfile(manifest) for manifest in manifests]):
-        raise NonValidInputError(f"Specified manifests {manifests} contain invalid file.")
+    file_name_manifest_dict: Dict[str, BaseAssetManifest] = _get_file_name_manifest_dict(
+        manifests=manifests
+    )
+    path_mapping_rule_list: List[PathMappingRule] = _process_path_mapping(
+        path_mapping_rules=path_mapping_rules
+    )
```

Comment:
nit on variable naming: maybe the parameter can be renamed to `path_mapping_rules_file`, then this variable with the processed rules can just be `path_mapping_rules`. How does that sound?